File: exercicio03/java/Fazenda2.java
package exercicio03.java;

public class Fazenda2 {
    public String nome;
    public double area;
    public String propietario;
    public String produto;
    private boolean adubado;
    private boolean milhoPlantado;

    public Fazenda2(String nome, double area, String propietario, String produto) {
        this.nome = nome;
        this.area = area;
        this.propietario = propietario;
        this.produto = produto;
    }

    public boolean isAdubado() {
        return adubado;
    }

    public void setAdubado(boolean adubado) {
        this.adubado = adubado;
    }

    public boolean isMilhoPlantado() {
        return milhoPlantado;
    }

    public void setMilhoPlantado(boolean milhoPlantado) {
        this.milhoPlantado = milhoPlantado;
    }

    public void adubar() {
        if (this.area > 0) {
            System.out.println("Área adubada");
            setAdubado(true);
        } else {
            System.out.println("É preciso ter áreas para plantar");
            setAdubado(false);
        }
    }

    public void plantar() {
        if (isAdubado()) {
            System.out.println(this.produto + "foram plantados(as)");
            this.milhoPlantado = true;
        }
    }

    public void imprimi() {
        System.out.println("Fazenda: " + this.nome);
        System.out.println("Area: " + this.area + " hectares");
        System.out.println("Proprietario: " + this.propietario);
        System.out.println("Produto: " + this.produto);
        System.out.print("Status da terra: ");
        if (isAdubado()) {
            System.out.println("adubado");
        } else {
            System.out.println("Não adubado");
        }
    }
}

/*
 * package atividad_ex03; public class Fazenda { public String nome; public
 * double area; public String propietario;
 * 
 * public Fazenda(String nome, double area, String propietario) { this.nome =
 * nome; this.area = area; this.propietario = propietario; }
 * 
 * public void plantar () { //metodo PLANTAR
 * System.out.println("Nova plantação adicionada a Fazenda " + this.nome); }
 * public void adubar () { //metodo ADUBAR
 * System.out.println("Adubando a nova plantação da fazenda " + this.nome); }
 * 
 * public void imprimi() { //para não ficar vazio criamos o metodo IMPRIMI
 * System.out.println("Fazenda: " + this.nome); System.out.println("Area: " +
 * this.area + " Hectares"); System.out.println("Proprietario: \n" +
 * this.propietario); } }
 */

File: exercicio03/python/fazenda2.py
class Fazenda2:
    nome = ""
    area = 0.0
    propietario = ""
    produto = ""
    adubado = False
    milho_plantado = False

    def __init__(self, nome: str, area: float, propietario: str, produto: str):
        self.nome = nome
        self.area = area
        self.propietario = propietario
        self.produto = produto
        self.adubado
        self.milho_plantado

    def is_adubado(self) -> bool:
        return self.adubado

    def set_adubado(self, adubado):
        self.adubado = adubado

    def is_milho_plantado(self):
        return self.milho_plantado

    def set_milho_plantado(self, milho_plantado):
        self.milho_plantado = milho_plantado

    def adubar(self):
        if (self.area > 0):
            print("Área adubada")
            self.set_adubado(True)
        else:
            print("É preciso ter áreas para plantar")

    def plantar(self):
        if self.is_adubado():
            print(f"{self.produto} foi plantado(a)")
            self.set_milho_plantado(True)
        else:
            print("Adube a terra para plantar")

    def imprimi(self):
        print("Fazenda:", self.nome)
        print("Área:", self.area, "a")
        print("Propietário:", self.propietario)
        print("Produto:", self.produto)
        print("Status da terra: ", end="")
        if self.is_adubado() and self.is_milho_plantado():
            print("adubado e plantado")
        else:
            print("não adubado ou nada plantado")
